lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -380,6 +386,42 @@ class ReportUIFeatures {
     this._document.body.removeChild(a);
     setTimeout(_ => URL.revokeObjectURL(href), 500);
   }
+
+  _handleStickyHeader() {
+    const topbarEl = this._document.querySelector('.lh-topbar');
+    const scoreScaleEl = this._document.querySelector('.lh-scorescale');
+    const stickyHeaderEl = this._document.querySelector('.lh-sticky-header');
+    const highlightEl = this._document.querySelector('.lh-highlighter');
+    if (!topbarEl || !scoreScaleEl || !stickyHeaderEl || !highlightEl) return;
+
+    // Show sticky header when the score scale begins to go underneath the topbar.
+    const showStickyHeader =
+      topbarEl.getBoundingClientRect().bottom - scoreScaleEl.getBoundingClientRect().top >= 0;
+    stickyHeaderEl.classList.toggle('lh-sticky-header--stuck', showStickyHeader);
+
+    // Highlight mini gauge when section is in view.
+    // Use the middle of the viewport as an anchor - the closest category to the middle
+    // is the one "in view".
+    let highlightIndex = 0;
+    let highlightIndexDistance = Number.POSITIVE_INFINITY;
+    const categoryEls = this._document.querySelectorAll('.lh-category');
+    for (const [index, categoryEl] of Object.entries(categoryEls)) {
+      // Normalize to middle of viewport.
+      const distanceToMiddle = categoryEl.getBoundingClientRect().top - window.innerHeight / 2;
+      // Closest negative distance to zero wins.
+      if (distanceToMiddle < 0 && highlightIndexDistance > distanceToMiddle) {
+        highlightIndex = parseInt(index);
+        highlightIndexDistance = -distanceToMiddle;
+      }
+    }
+
+    // Category order matches gauge order in sticky header.
+    // TODO(hoten): not 100% true yet, need to order gauges like: core, pwa, plugins. Remove
+    // this comment when that is done.
+    const gaugeToHighlight = stickyHeaderEl.querySelectorAll('.lh-gauge__wrapper')[highlightIndex];
+    // @ts-ignore
```

Comment:
what are we ignoring?

Reply:
`Property 'style' does not exist on type 'Element'` idk why

Reply:
needs to be HTMLElement